presto-spark-launcher/src/main/java/com/facebook/presto/spark/launcher/PrestoSparkLoader.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.launcher;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+// Based on PluginClassLoader
+public class PrestoSparkLoader
```

Comment:
curious if we can unify `PrestoSparkLoader` and `PluginClassLoader` ? :) . Have two class loader which is very similar might be difficult to maintain.

Reply:
To have them in a common place a common module is needed. I was trying to avoid having PrestoSparkLauncher to depend on any Presto or Airlift modules to avoid any potential dependency clashes. 

I wouldn't expect this class to get changed frequently, thus i think having a "copy" of it is lesser of two evils.